nmclicktests/environment.js

```diff
@@ -193,107 +20,54 @@ const prependLines = (pre, text) => {
 // Environment provides
 class Environment {
   constructor(config) {
-    this.nm = Nightmare({ Promise: Bluebird, typeInterval: 500, show: false });
     this.config = config || {};
     this.config.rootPath = (typeof this.config.rootPath === 'string') ? this.config.rootPath : '';
-    this.config.serverTimeout = this.config.serverTimeout || 15000;
+    this.config.serverTimeout = this.config.serverTimeout || 35000;
     this.config.viewWidth = 2000;
     this.config.viewHeight = 2000;
     this.config.showServerOutput = this.config.showServerOutput === undefined ? true : this.config.showServerOutput;
     this.config.serverStartupOptions = this.config.serverStartupOptions || [];
     this.shuttinDown = false;
-
-    // init
-    this.nm.viewport(this.config.viewWidth, this.config.viewHeight);
-    this.nm.on('console', (type, msg1, msg2) => {
-      winston.info(`[ui ${type}] ${(new Date()).toISOString()}  - ${msg1} ${JSON.stringify(msg2)}`);
-
-      if (type === 'error' && !this.shuttinDown) {
-        winston.info('ERROR DETECTED!');
-      }
-    })
   }
 
-  getRootUrl() { return rootUrl; }
+  getRootUrl() { return this.rootUrl; }
 
   getPort() {
     const tmpPortrange = portrange + Math.floor((Math.random() * 5000));
 
-    return new Bluebird((resolve, reject) => {
+    return new Promise((resolve, reject) => {
       const server = net.createServer();
 
-      server.listen(tmpPortrange, (err) => {
+      server.listen(tmpPortrange, () => {
         server.once('close', () => {
           this.port = tmpPortrange;
-          rootUrl = `http://localhost:${this.port}${this.config.rootPath}`
+          this.rootUrl = `http://localhost:${this.port}${this.config.rootPath}`;
           resolve();
         });
         server.close();
       });
-      server.on('error', (err) => {
+      server.on('error', () => {
         this.getPort().then(resolve);
       });
     });
   }
 
-  ensureStarted() {
-    return Bluebird.resolve()
-      .then(() => {
-        if (!this.hasStarted) {
-          return Bluebird.resolve()
-            .delay(50)
-            .then(() => this.ensureStarted());
-        }
+  async init() {
+    try {
+      this.browser = await puppeteer.launch({
+        defaultViewport: {
+          width: this.config.viewWidth,
+          height: this.config.viewHeight
+        },
+        headless: true
```

Comment:
Change this to `false` to see the browser!

Reply:
Would it makes sense to make that configurable for local runs?

Reply:
Yes it would be nice, I use to set a `GUI=1` env variable for this

Reply:
Yes, we could make them configurable via environment variables and document them because otherwise they are as hidden as this boolean.

I usually don't use environment variables to configure stuff like this, is `GUI=1` unique enough? Would it clash with some other programs? What about `GUI=true`?